docs/includes/widgets/common.md

```diff
@@ -0,0 +1,46 @@
+# Widget_Common - `common.php`
+
+## Technical Description:
+
+This class is responsible for registering the controls under the `Advanced` tab of each widget in Elementor.
+
+
+## Features:
+
+### Mask:
+We've added a mask option to any type of widget in Elementor, using the CSS `mask` property.
+
+#### Knowledge Base:
+- [Mask Option on Elementor.com](https://elementor.com/help/mask-option/)
+  
+
+- [Mask property reference on MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/mask)
+
+
+- [CSS-Tricks: Clipping and Masking in CSS](https://css-tricks.com/clipping-masking-css/)
+  
+
+- [CSS-Tricks: mask-image](https://css-tricks.com/almanac/properties/m/mask-image/)
+  
+
+- [web.dev: Apply effects to images with CSS's mask-image property](https://web.dev/css-masking/)
+  
+
+- [CSS & SVG Masks](https://lab.iamvdo.me/css-svg-masks/)
+
+#### Functions Reference:
+
+- `get_shapes( $add_custom )` - Returns a translated user-friendly list of the available masking shapes.
+
+  Passing `true` (which is the default) as a parameter will also append a 'Custom' option to the list in order to use with a `SELECT` control type.
+
+  Example use-case: Letting the user select a `custom` option, and show an image upload control.
+
+
+- `get_mask_selectors( $rules )` - Gets a string of CSS rules to apply, and returns an array of selectors with those rules.
+
+	This function has been created in order to deal with masking for image widget. For most of the widgets we mask the wrapper itself, 
+  
+	but in the case of an image widget, we wanted to mask the `img` tag directly. So instead of writing a lot of selectors every time, 
+  
+	we created a function that builds both of those selectors for us.
```

Comment:
Please move all "Functions Reference" to the code itself. 
Here is recommended to describe some decisions and constraints that you faced during the development

Reply:
@StyleShit did you moved the "Functions Reference" to the code itself?

Reply:
It already contains PHPDoc

Reply:
Here you extended the description, for example:

![image](https://user-images.githubusercontent.com/21468434/120690503-f85a8400-c4ad-11eb-8121-fe081cbe9c42.png)

VS

![image](https://user-images.githubusercontent.com/21468434/120690441-e4af1d80-c4ad-11eb-8651-68116223f019.png)

Reply:
Done.